src/main/java/bhs/devilbotz/commands/arm/ArmToPosition.java

```diff
@@ -3,85 +3,33 @@
 import bhs.devilbotz.Constants.ArmConstants;
 import bhs.devilbotz.subsystems.Arm;
 import bhs.devilbotz.subsystems.Gripper;
-import edu.wpi.first.wpilibj2.command.CommandBase;
 
-public class ArmToPosition extends CommandBase {
-  private final Arm arm;
+public class ArmToPosition extends ArmSafety {
   private final double targetPosition;
-  private final double targetPositionTolerance;
 
-  private Gripper gripper;
-  private double gripperClosePosition;
-  private boolean bReachedLimit;
-
-  public ArmToPosition(Arm arm, double targetPosition) {
-    this.arm = arm;
+  public ArmToPosition(Arm arm, double targetPosition, Gripper gripper) {
+    super(arm, gripper);
     this.targetPosition = targetPosition;
-    targetPositionTolerance = ArmConstants.POSITION_TOLERANCE;
-    bReachedLimit = false;
-    gripper = null;
-    gripperClosePosition = 0;
-    addRequirements(arm);
-  }
-
-  public ArmToPosition(
-      Arm arm, double targetPosition, Gripper gripper, double gripperClosePosition) {
-    this(arm, targetPosition);
-    this.gripper = gripper;
-    this.gripperClosePosition = gripperClosePosition;
-    addRequirements(gripper);
   }
 
-  // Called when the command is initially scheduled.
-  @Override
-  public void initialize() {}
-
   // Called every time the scheduler runs while the command is scheduled.
   @Override
-  public void execute() {
-    double currentPosition = arm.getPosition();
+  public void executeWithSafety() {
+    double currentPosition = getPosition();
 
     if (currentPosition < targetPosition) {
-      if (arm.isTopLimit()) {
-        bReachedLimit = true;
-      } else {
-        if ((null != gripper) && arm.isBottomLimit()) {
-          // We want to close the gripper if it is at the bottom and want to move up
-          gripper.close();
-        }
-        arm.up();
-      }
+      up();
     } else if (currentPosition > targetPosition) {
-      if (arm.isBottomLimit()) {
-        bReachedLimit = true;
-      } else {
-        arm.down();
-        if ((null != gripper) && (currentPosition < gripperClosePosition)) {
-          gripper.close();
-        }
-      }
-    }
-
-    if (bReachedLimit) {
-      arm.stop();
+      down();
```

Comment:
Will this cause oscillation?

Reply:
It shouldn't because in `isFinishedWithSafety()`, there's a pass tolerance window.  If the arm moves fast enough such that the position changes by more than POSITION_TOLERANCE, then it _could_ oscillate, I suppose.

```
    return (currentPosition >= targetPosition)
        && (currentPosition < targetPosition + targetPositionTolerance);
        && (currentPosition < targetPosition + ArmConstants.POSITION_TOLERANCE);
```